devicetypes/smartthings/fibaro-rgbw-controller.src/fibaro-rgbw-controller.groovy

```diff
@@ -26,562 +26,233 @@
-
-     if (cmd.value >= 5) {
-          def item2 = new LinkedHashMap(item1)
-          item2.name = "level"
-          item2.value = (cmd.value == 99 ? 100 : cmd.value) as String
-          item2.unit = "%"
-          item2.descriptionText = "${item1.linkText} dimmed ${item2.value} %"
-          item2.canBeCurrentState = true
-          item2.isStateChange = isStateChange(device, item2.name, item2.value)
-          item2.displayed = false
-          result << item2
-     }
-     result
+def handleMeterReport(cmd){
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kWh")
+		} else if (cmd.scale == 1) {
+			createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kVAh")
+		} else if (cmd.scale == 2) {
+			createEvent(name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W")
+		}
+	}
 }
 
-def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd, item1) {
-	log.debug "${device.displayName} parameter '${cmd.parameterNumber}' with a byte size of '${cmd.size}' is set to '${cmd.configurationValue}'"
-}
-/*
-def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
-     log.debug "Report: $cmd"
-     def value = "when off"
-     if (cmd.configurationValue[0] == 1) {value = "when on"}
-     if (cmd.configurationValue[0] == 2) {value = "never"}
-     [name: "indicatorStatus", value: value, displayed: false]
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+	log.debug "v3 Meter report: "+cmd
+	handleMeterReport(cmd)
 }
-*/
-def createEvent(physicalgraph.zwave.Command cmd,  Map map) {
-     // Handles any Z-Wave commands we aren't interested in
-     log.debug "UNHANDLED COMMAND $cmd"
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	 // Handles any Z-Wave commands we aren't interested in
+	log.debug "UNHANDLED COMMAND $cmd"
 }
 
 def on() {
 	log.debug "on()"
-	sendEvent(name: "switch", value: "on")
-	delayBetween([zwave.basicV1.basicSet(value: 0xFF).format(),
-    			  zwave.switchMultilevelV1.switchMultilevelGet().format()], 5000)
+	commands([zwave.basicV1.basicSet(value: 0xFF),
+				zwave.switchMultilevelV3.switchMultilevelGet(),
+				zwave.meterV3.meterGet(scale: 2)], 5000)
 }
 
 def off() {
 	log.debug "off()"
-	sendEvent(name: "switch", value: "off")
-	delayBetween ([zwave.basicV1.basicSet(value: 0x00).format(), zwave.switchMultilevelV1.switchMultilevelGet().format()], 5000)
+	commands([zwave.basicV1.basicSet(value: 0x00),
+				zwave.switchMultilevelV3.switchMultilevelGet(),
+				zwave.meterV3.meterGet(scale: 2)], 5000)
 }
 
 
-def poll() {
-    zwave.switchMultilevelV1.switchMultilevelGet().format()
+def ping() {
+	refresh()
 }
 
 def refresh() {
-	def cmd = []
-	cmd << response(zwave.switchMultilevelV1.switchMultilevelGet().format())
-    delayBetween(cmd, 500)
+	commands([zwave.switchMultilevelV1.switchMultilevelGet(), zwave.meterV3.meterGet(scale: 2)])
```

Comment:
While the same, we should use one Version of command classes, so either use V3 or V1